gsfpy/__init__.py

```diff
@@ -1,173 +1,62 @@
-
-    def __enter__(self):
-        return self
-
-    def __exit__(self, exc_type, exc_val, exc_tb):
-        self.close()
-
-    @property
-    def file_mode(self) -> FileMode:
-        """
-        Mode the file has been opened in
-        """
-        return self._file_mode
-
-    def close(self):
-        """
-        Once this method has been called further operations will fail
-        :raises GsfException: Raised if anything went wrong
-        """
-        _handle_failure(gsfClose(self._handle))
-
-    def seek(self, option: SeekOption):
-        """
-        :param option: Where to seek to
-        :raises GsfException: Raised if anything went wrong
-        """
-        _handle_failure(gsfSeek(self._handle, option))
-
-    def read(
-        self,
-        desired_record: RecordType = RecordType.GSF_NEXT_RECORD,
-        record_number: int = 0,
-    ) -> Tuple[c_gsfDataID, c_gsfRecords]:
-        """
-        When the file is open in GSF_READONLY_INDEX or GSF_UPDATE_INDEX mode then the
-        record_number parameter may be used to indicate which instance of the record to
-        read.
-        :param desired_record: Record type to read
-        :param record_number: nth occurrence of the record to read from, starting from 1
-        :return: Tuple of c_gsfDataID and c_gsfRecords
-        :raises GsfException: Raised if anything went wrong
-        """
-        data_id = c_gsfDataID()
-        data_id.record_number = record_number
-
-        records = c_gsfRecords()
 
-        _handle_failure(
-            gsfRead(self._handle, desired_record, byref(data_id), byref(records))
-        )
-
-        return data_id, records
-
-    def write(
-        self, records: c_gsfRecords, record_type: RecordType, record_number: int = 0
-    ):
-        """
-        When the file is open in GSF_UPDATE_INDEX mode then the record_number parameter
-        may be used to indicate which instance of the record to write over.
-        :param records: Data to write
-        :param record_type: Specifies the type of record to write to
-        :param record_number: nth occurrence of the record to write to, starting from 1
-        :raises GsfException: Raised if anything went wrong
-        """
-        data_id = c_gsfDataID()
-        data_id.recordID = record_type
-        data_id.record_number = record_number
-
-        _handle_failure(gsfWrite(self._handle, byref(data_id), byref(records)))
-
-    def get_number_records(self, desired_record: RecordType) -> int:
-        """
-        May only be used when the file is open for direct access (GSF_READONLY_INDEX or
-        GSF_UPDATE_INDEX).
-        :param desired_record: Specifies the type of record to count
-        :return: Number of records of type desired_record, otherwise -1
-        """
-        count = gsfGetNumberRecords(self._handle, desired_record)
-        _handle_failure(count)
-        return count
+def mirror_default_gsf_version_submodule(_globals: dict, submodule_name: str = None):
```

Comment:
Perhaps rename the _globals to globals_dict.  Leading underscore is misleading as it used for privacy etc.  In this case I believe the _ is used to avoid clashing with the globals command?

Reply:
Done, yes the original name was used to avoid the clash with globals(). Agree that globals_dict is a better alternative.